### ISSUE TITLE
Bucket status on longer includes the ARN, it did previously at some point. Docs also state that it's supposed to.

### ISSUE THREAD

**Issue by jbilliau-rcd** [aws-controllers-k8s/community#2094]
**Describe the bug**

While testing version `1.0.10` of the ack-s3-controller, I noticed the `Bucket` object no longer has the ARN of the created bucket in its status field no more. According to the docs, it's supposed to, and I know it used to when I was playing with a previous version. https://aws-controllers-k8s.github.io/community/reference/s3/v1alpha1/bucket/

```
status:
  ackResourceMetadata:
    ownerAccountID: '414323852230'
    region: us-east-2
  conditions:
    - lastTransitionTime: '2024-06-19T12:49:17Z'
      message: Resource synced successfully
      reason: ''
      status: 'True'
      type: ACK.ResourceSynced
```

**Steps to reproduce**

Create a `Bucket` object, note the status.

**Expected outcome**

`Bucket` has the ARN of the bucket it it's status, and thus can be used for `FieldExport`

* Kubernetes version: 1.28
* Using EKS (yes/no), if so version?  Yes
* AWS service targeted (S3, RDS, etc.) S3

**Comment by jbilliau-rcd** [comment]
I see this issue here https://github.com/aws-controllers-k8s/community/issues/270

@vijtrip2 does the S3 create bucket API call no longer produce the ARN? Is that why this no longer works?

**Comment by a-hilaly** [comment]
@jbilliau-rcd great catch, i just re-opened #270.

The S3 API doesn't return any Bucket ARN, I guess we can construct one for S3 Buckets..

**Comment by jbilliau-rcd** [comment]
That would be great, thank you @a-hilaly . Just started diving into ACK and building out Helm charts for our developers to start deploying their AWS resources with it. My end-game goal is to use `FieldExports` to dump all relevant ARN's from all creates resources (SQS, S3, DynamoDB, etc.) to a central `ConfigMap` so that they have a one-stop shop to see all of their created AWS resources ARN's. That's when I noticed this was missing.

**Comment by jbilliau-rcd** [comment]
Looks like this was fixed in 1.0.15, thanks @a-hilaly !
